server/app/src/test/java/io/whitefox/api/deltasharing/server/DeltaSharesApiImplAwsTest.java

```diff
@@ -232,4 +232,133 @@ public void queryTableByVersion() throws IOException {
     assertEquals(7, responseBodyLines.length);
     assertEquals(s3DeltaTable1FilesWithoutPresignedUrl, files);
   }
+
+  @DisabledOnOs(OS.WINDOWS)
+  @Test
+  public void queryIcebergTableCurrentVersion() throws IOException {
+    var responseBodyLines = given()
+        .when()
+        .filter(deltaFilter)
+        .body("{}")
+        .header(new Header("Content-Type", "application/json"))
+        .post(
+            "delta-api/v1/shares/{share}/schemas/{schema}/tables/{table}/query",
+            "s3share",
+            "s3schema",
+            "s3IcebergTable1")
+        .then()
+        .statusCode(200)
+        .extract()
+        .body()
+        .asString()
+        .split("\n");
+
+    assertEquals(
+        s3IcebergTable1Protocol,
+        objectMapper.reader().readValue(responseBodyLines[0], ParquetProtocol.class));
+    assertEquals(
+        s3IcebergTable1Metadata,
+        objectMapper.reader().readValue(responseBodyLines[1], ParquetMetadata.class));
+    var files = Arrays.stream(responseBodyLines)
+        .skip(2)
+        .map(line -> {
+          try {
+            return objectMapper
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                .reader()
+                .readValue(line, FileObjectWithoutPresignedUrl.class);
+          } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Even if this is a test case? I mean, this exception should never be thrown but it's mandatory since we are using the object mapper. 

Do you think we need to handle it better? We need to change it in all tests

Reply:
Runtime exception is fine it's the message that is missing, I will add it :)